apps/code-examples/src/app/code-examples/router/href/basic/sky-href-demo.component.spec.ts

```diff
@@ -1,70 +1,70 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { By } from '@angular/platform-browser';
-import { SkyHrefModule, SkyHrefResolverService } from '@skyux/router';
+import { SkyHrefHarness, SkyHrefTestingModule } from '@skyux/router/testing';
 
-import { MockResolverService } from './fixtures/mock-resolver-service';
 import { SkyHrefDemoComponent } from './sky-href-demo.component';
 
 describe('SkyHrefDemoComponent', () => {
   let component: SkyHrefDemoComponent;
   let fixture: ComponentFixture<SkyHrefDemoComponent>;
-  let resolverService: MockResolverService;
-
-  async function setup(userHasAccess: boolean): Promise<void> {
-    resolverService = new MockResolverService(userHasAccess);
 
+  async function setup(options: { userHasAccess: boolean }) {
     TestBed.configureTestingModule({
       declarations: [SkyHrefDemoComponent],
-      imports: [SkyHrefModule],
-      providers: [
-        {
-          provide: SkyHrefResolverService,
-          useValue: resolverService,
-        },
+      imports: [
+        SkyHrefTestingModule.with({ userHasAccess: options.userHasAccess }),
       ],
     });
 
     fixture = TestBed.createComponent(SkyHrefDemoComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
-    return fixture.whenStable().then();
+    await fixture.whenStable();
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    return { fixture, loader };
   }
 
   it('should create', async () => {
-    await setup(true);
+    await setup({ userHasAccess: true });
     expect(component).toBeTruthy();
   });
 
   it('should show skyhref with access', async () => {
-    await setup(true);
-    expect(fixture.debugElement.query(By.css('#allow'))).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#allow')).attributes['href']
-    ).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#allow')).attributes['hidden']
-    ).toBeFalsy();
+    const { loader } = await setup({ userHasAccess: true });
+    const hrefHarness = await loader.getHarness(
```

Comment:
I wonder if you could move the `getHarness` logic to the `setup` function as we've done for the other harnesses? https://github.com/blackbaud/skyux/blob/main/apps/code-examples/src/app/code-examples/indicators/alert/basic/alert-demo.component.spec.ts#L18-L20 
You'd have to pass in the `dataSkyId` option, however.

Reply:
Moved `getHarness` call to setup.